### ISSUE TITLE
Add ability to show parallel coordinates on HPO run

### ISSUE THREAD

**Issue by Samreay** [allegroai/clearml#279]
having the parallel coordinates plot available from the HP parent task. Right now, if I want to see the parallel coord plot (shown below), I have to manually select all trials in a HPO run > compare > Hyperparameters > Parallel Coordinates > Select metric > Select parameters. However, as the HPO task already knows what metric we’re optimising and that the hyperparameters being optimised are, it would be so much nicer to be able to click on the HPO parent task and have it available. (In the HPO > Results > Plots tab there is a summary table, so essentially this plotted in an informative manner).

![image](https://user-images.githubusercontent.com/6748505/103912385-24d8c280-50ff-11eb-9baf-3ce770fae38d.png)

**Comment by bmartinn** [comment]
Hi @Samreay 

A bit of background before diving into possible designs / solutions.
The Parallel coordinates is designed as a way to compare any Task in the system. The data itself is fetched from the backend server and processed by the UI, then display here. Unfortunately on a per Task basis the system does not yet have "custom" widget, generally speaking it uses the builtin visualization capabilities of plotly. That basically means that taking a component from the comparison page and customizing it to fit into a Task API is far from trivial.

With the above in mind, where exactly is the pain we are trying to solve? Is it the need to manually select the Tasks the HPO process created?
Will it be easier if we had a quick way to select all these Tasks for comparison? 
(just throwing it out there, select all child Tasks button, as all the HPO tasks are "children" of the parent HPO process)

WDYT?

**Comment by Samreay** [comment]
Ah I didnt realise there was a select all children button, which simplifies one of the steps, but you then have to build up the parallel coordinate plot yourself, which will become super tedious as the number of parameters grows above one or two. The odd thing for me is that this information is already in the parent task in the summary table. Its just far easier to parse trends in a table if we can see the plots visually - thats when we know whats correlating with what and which hyperparameters are the most sensitive.

**Comment by bmartinn** [comment]
> Ah I didnt realise there was a select all children button, which simplifies one of the steps

Actually there isn't, I was just suggesting it as an alternative solution to the problem :)

On the Task itself we can do something like the last plot [here](https://plotly.com/javascript/parallel-coordinates-plot/)

WDYT?

**Comment by Samreay** [comment]
Yeah that's precisely what I have coded up manually and uploaded as a
plotly artifact, I can just see it being useful to be there automatically.

**Comment by bmartinn** [comment]
@Samreay Nice!
Any chance you want to PR the code if you have already coded it ?

**Comment by Samreay** [comment]
I could PR something that uploads a plotly artifact, but nothing that
actually integrates into the UI properly

**Comment by bmartinn** [comment]
No worries, I'll make sure we take it from there :)

**Comment by Samreay** [comment]
Hey mate, I was going to say Im not too sure how youd actually want the code structures, or in what class.

This is the snippet I had, which Im 100% sure can be improved upon:

```
def get_parallel(experiments, param_group="General", metric_group="evaluate"):
    comparison = []
    for e in experiments:
        row = {"id": e.id, **e.get_parameters_as_dict()[param_group]}
        row.update({k: v["last"] for k, v in e.get_last_scalar_metrics()[metric_group].items()})
        comparison.append(row)
    df = pd.DataFrame(comparison)
    df = df.set_index("id")
    fig = px.parallel_coordinates(df)
    return fig


fig = get_parallel(an_optimizer.get_top_experiments(999))
fig.show()
```

**Comment by bmartinn** [comment]
Thanks @Samreay !
This is exactly what I needed, let me see what I can do ;)

**Comment by bmartinn** [comment]
Hey @Samreay checkout v 0.17.4 ;)

![newplot (13)](https://user-images.githubusercontent.com/51887611/104520233-6c7db380-5603-11eb-9dd0-87811a12311d.png)


``` bash
pip install clearml==0.17.4
```

**Comment by Samreay** [comment]
This is a thing of pure beauty. You are a legend.

**Comment by mhashas** [comment]
Hi, has something happened to this? All our colors are green
![image](https://user-images.githubusercontent.com/13119609/128830094-1daf32e3-f22b-4905-95f2-35710907614b.png)

**Comment by jkhenning** [comment]
Hi @mhashas ,

Which Server version are you using?

**Comment by bmartinn** [comment]
@mhashas it should be fixed with `clearml-server` 1.1.1, can you verify ?

**Comment by jkhenning** [comment]
Closing this as it was already released. Please reopen if required.
